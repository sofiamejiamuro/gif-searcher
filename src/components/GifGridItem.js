import React from 'react'

const GifGridItem = ({ title, url }) => {
  
  return (
    <div className="card">
      <p>{ title }</p>
      <img src={ url } alt={ title }></img>     
    </div>
  )
}

export default GifGridItem;