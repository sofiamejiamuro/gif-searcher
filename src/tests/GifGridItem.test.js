import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import GifGridItem from '../components/GifGridItem';

describe('Pruebas en componente GifGridItem', () => {

  const title = 'Ranma';
  const url= 'https://algunacosa';
  const wrapper = shallow(<GifGridItem 
    title={ title }
    url={ url }  
    />);

  test('Debe de mostrar el componente correctamente y sus proptypes', () => {

    // Wrapper commonly used in rendered components
    // Shallow is a method from enzyme, similar to render from react but with more options
    expect( wrapper ).toMatchSnapshot();
    
  })
  
  test('Debe de tener un pÃ¡rrafo con el texto del props', () => {
    
    const p = wrapper.find('p').text().trim();

    expect(p).toBe('Ranma')

  })
  
  test('Debe de obtener una imagen con atributos url y title igual a los props', () => {

    const img = wrapper.find('img');
    console.log(img.html());
    console.log(img.props());
    console.log(img.props().src);
    
    expect(img.props().src).toBe(url);
    expect(img.props().alt).toBe(title);

  })

  test('Debe tener la clase animate__rubberBand ', () => {

    const div = wrapper.find('div');
    expect(div.props().className.includes('animate__rubberBand')).toBe(true)

  })
  
  
})
