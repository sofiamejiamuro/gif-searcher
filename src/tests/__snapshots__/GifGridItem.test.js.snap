// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente GifGridItem Debe de mostrar el componente correctamente y sus proptypes 1`] = `
<div
  className="card animate__animated animate__rubberBand"
>
  <p>
    Ranma
  </p>
  <img
    alt="Ranma"
    src="https://algunacosa"
  />
</div>
`;
